controllers/ItemController.js

```diff
@@ -1,18 +1,169 @@
+const { deleteModel } = require("mongoose");
 const itemModel = require("../models/itemModel");
 
 const controllers = {
     createItem: (req, res) => {
-        itemModel.create({
-            title: req.body.title,
-            postType: req.body.type,
-            description: req.body.description,
-            category: req.body.category,
-            images: req.body.images,
-            delivery: req.body.delivery,
-            status: req.body.status,
-            tags: req.body.tags,
-            postedBy: req.body.postedBy, //to review in frontend
-        });
+        console.log(req.body);
+        itemModel
+            .create({
+                title: req.body.title,
+                postType: req.body.type,
+                description: req.body.description,
+                category: req.body.category,
+                images: req.body.images,
+                delivery: req.body.delivery,
+                status: req.body.status,
+                tags: req.body.tags,
+                postedBy: req.body.postedBy, //to review in frontend
+            })
+            .then((result) => {
+                res.json({
+                    success: true,
+                    message: "item successfully created",
+                });
+            })
+            .catch((err) => {
+                console.log(err);
+                res.statusCode = 500; // to check
+                res.json({
+                    success: false,
+                    message: "create item failed",
+                });
+            });
+    },
+    // getItem: (req, res) => {
+    //     res.json("get item");
+    // },
+
+    updateItem: (req, res) => {
+        console.log(req.params.id);
+        // to improve code later
+
+        itemModel
+            .findOne({
+                _id: req.params.id,
+            })
+            .then((result) => {
+                if (!result) {
+                    res.statusCode = 404;
+                    res.json({
+                        success: false,
+                        message: "No item with that ID found",
+                    });
+                    return;
+                }
+
+                // tidy code later
+                const itemUpdates = {
+                    title: req.body.title,
+                    postType: req.body.type,
+                    description: req.body.description,
+                    category: req.body.category,
+                    images: req.body.images,
+                    delivery: req.body.delivery,
+                    status: req.body.status,
+                    tags: req.body.tags,
+                    postedBy: req.body.postedBy, //to review in frontend
+                };
+
+                for (const prop in itemUpdates) {
+                    if (!itemUpdates[prop]) {
+                        delete itemUpdates[prop];
```

Comment:
if user updates a field to be an empty string, !itemUpdates[prop] would evaluate to true and the field would be deleted from the update object. This means that any deletion of content won't be updated

Reply:
Updated code to check for empty fields & validate against schema too